labs/3L.md

```diff
@@ -0,0 +1,108 @@
+### OR
+### Using Command Palate
+1. Open the command palate with `Ctrl + Shift + P` on Windows or `Cmd + Shift + P` on Mac.
+2. Type "run task" and select "Tasks: Run Task" (or "Tasks: Rerun Last Task" in the future).
+3. Select the "Grade Assignment" task. Use the other ones as needed but avoid the "Update" task unless told to do so.
+4. Select `script.rhai` and press Enter.
+
+### [WATCH: Checking for Compiler Errors.](<Link_todo>)
+## Project Overview
+Here are the files in your project for the ConcertTicket package:
+
+```
+.
+├── src/
+│   └── ConcertTicket/
+│       ├── ConcertTicket.java
+│       └── Wallet.java
+└── test/
+    └── ConcertTicket/
+        └── WalletTest.java
+```
+
+Using the files tab on the left, double-click on the `ConcertTicket.ConcertTicket` class and the `ConcertTicket.Wallet` class.![image.png](<Link_todo>)
+
+Get familiar with these files and read through the comments!
+
+## Implementation Instructions
+
+1. **Implement `getSize()` Method**
+    - The size is the number of tickets currently present in the wallet.
+
+2. **Implement `getLength()` Method**
+    - The length is the capacity of the wallet - the maximum number of tickets our wallet can hold.
+
+3. **Implement `remove` Method**
+    - There is no method for removing a `ConcertTicket` from the wallet! Your job is to write that method. The method should be called `remove`. It should take no parameters and it should return the most recently added `ConcertTicket`.
+    ```java
+    public ConcertTicket remove() {
+        // Create a ConcertTicket reference variable (DO NOT create a
+        // new ConcertTicket - JUST create the reference variable)
+
+        // Use the size variable (which always points at the next empty
+        // slot) to get the last added ConcertTicket from the array:
+
+        // Set that array slot to null:
+
+        // Decrement the size variable:
+
+        // Return the Concert Ticket:
+    }
+    ```
+
+4. **Checking for Errors**
+    - Once you are done, make sure there are no syntax errors. You can do so by opening the problems view by clicking on this button in the bottom left:
+
+        ![Problems View Button](<Link_todo>)
+
+    - Which opens this tab:
+
+        ![Problems View Tab](<Link_todo>)
+
+5. **Running Tests**
+    - If there are no errors, go to `WalletTest.java` in the `test/ConcertTicket` folder. Click on the thick double green arrow on line 7 to run all tests-
+
+        ![Double green arrow](<Link_todo>)
+
+6. **Submitting**
+    - If all tests have passed, hit submit!
+
+## Rubric
+- **Project must compile to receive a grade.**
+- **All tests for `ConcertTicket.WalletTest` must pass (25 points).**
+
+## Submission
+
+1. You can copy and run the `umm create-submission` command in the terminal, and that should create a zip file with a name similar to `submission-2024-01-24-15-04-50.zip`.
+    ```bash
+    umm create-submission
+    ```
+
+2. Submit the `submission-2024-... .zip` file to Gradescope.
```

Comment:
add instructions on how to find the file.

... the submission zip will appear in the file explorer tab in vscode. They can right click on this file and click on reveal in explorer (windows) or reveal in finder (mac) in order to find this file. then, they can drag and drop this to gradescope for submission.